src/environment/stella_environment.cpp

```diff
@@ -212,7 +212,11 @@ const ALEState& StellaEnvironment::getState() const {
 }
 
 void StellaEnvironment::saveScreenPNG(const string& filename) {
-  m_osystem->p_export_screen->save_png(m_screen, filename);
+
```

Comment:
The ScreenExporter constructor is actually quite fast, and there's no memory being allocated here -- it's all on the stack. I wouldn't be too concerned about it -- it's only a few cycles compared to the cost of writing a full PNG to disk; and this way it's a bit cleaner.